### PR TITLE
Add plant to garden#7

### LINKED ISSUE #7: Add Plant to Garden
From my garden's show page, I can click a link to add a plant to my garden.  When I click this link, I am redirected to a create plant page.  On this page, I can add a plant name and times per week.  After I select 'create plant', I am redirected to my garden show page and I see a flash message that says, "Party in my plants, you've created a new plant!"

### PR DESCRIPTION
**Pull request by asmolentzov**
User Story 7: Add Plant to Garden
Builds on create_garden branch to be able to add a new plant to a garden. 
- Adds new plant route, controller, view, and form
- Adds plants routes as nested routes under gardens
- Adds flash messages to application.html.erb
- Adds error checking for creating a plant with blank info
- All tests passing

Closes #7

### CODE REVIEW

File: app/models/garden.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Garden < ApplicationRecord
+  belongs_to :user
+  has_many :plants
+  validates_presence_of :name, :zip_code
+  accepts_nested_attributes_for :plants, reject_if: :all_blank
```
Comment (le3ah) [inline app/models/garden.rb]: Oh, I like the `reject_if` command!

File: config/routes.rb [hunk 2]
```diff
@@ -1,3 +1,8 @@
 Rails.application.routes.draw do
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+
+  get '/dashboard', to: "users#show"
+  resources :gardens, only: [:new, :create, :show], shallow: true do
```
Comment (le3ah) [inline config/routes.rb]: nice use of `shallow`

File: app/models/garden.rb (hunk 1 above)
Comment (abenetka) [reply app/models/garden.rb]: These nested attributes and reject_if are so fancy and cool

File: spec/features/user_creates_garden_spec.rb [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+
+    expect(current_path).to eq(garden_path(garden))
+    expect(page).to have_content(garden.name)
+    expect(page).to have_content(garden.zip_code)
+    within ".plants" do
+      expect(page).to have_content("Roses")
+      expect(page).to have_content("Watering Requirements: 3 times/week")
+      expect(page).to have_content("Sunflowers")
+      expect(page).to have_content("Watering Requirements: 2 times/week")
+      expect(page).to have_content("My Plant: Fuzzy")
```
Comment (abenetka) [inline spec/features/user_creates_garden_spec.rb]: Nice plant  name

File: app/models/garden.rb (hunk 1 above)
Comment (asmolentzov) [reply app/models/garden.rb]: I know right? I was worried it might not work so well for individual plant create, but it seems to work just fine!

**Comment by asmolentzov** [comment]
Merge conflicts resolved and all tests passing!

**Pull request merged**